app/src/main/java/com/droidhats/mapprocessor/AStar.kt

```diff
@@ -117,37 +117,18 @@ fun findNearestPoint(mapElement: MapElement, pathElements: List<Vertex>): Vertex
 
 /**
  * Takes 2 map elements and returns the shortest path in between them given the list of path elements
+ * as an arrow
  * @param start element
  * @param end element
  * @param list of vertices
  * @return string
  */
-fun A_Star(start: MapElement, end: MapElement, pathElements: MutableList<Vertex>): String {
+fun getPath(start: MapElement, end: MapElement, pathElements: MutableList<Vertex>): String {
     val startVertex = findNearestPoint(start, pathElements)
     val endVertex = findNearestPoint(end, pathElements)
-    val queue = PriorityQueue()
 
-    // A* algorithm part
-    var poppedNode: Vertex? = startVertex
-    while (poppedNode != endVertex) {
-        for (neighbor in poppedNode!!.neighbors) {
-            if (neighbor.getValue() == null || poppedNode.getValue()!! < neighbor.getValue()!!) {
-                neighbor.setValue(getDistance(poppedNode.pos, neighbor.pos))
-                neighbor.prev = poppedNode
-                if (!queue.isNotWithin(neighbor)) {
-                    queue.removeVertex(neighbor)
-                }
-            }
-            neighbor.removeFromNeighbors(poppedNode)
-            if (queue.isNotWithin(neighbor)) {
-                queue.insert(neighbor)
-            }
-        }
-        poppedNode = queue.pop()
-        if (poppedNode == null) {
-            return A_Star(start, end, removeStartAndEnd(startVertex, endVertex, pathElements))
-        }
-    }
+    val path = aStar(startVertex, endVertex, start, end, pathElements)
+    if (path != null) return path
```

Comment:
Can we add parentheses to this if statement, or possibly move the return statement to a new line with proper indentation.